libcst/matchers/_visitors.py

```diff
@@ -53,12 +54,23 @@
 _CSTNodeT = TypeVar("_CSTNodeT", bound=cst.CSTNode)
 
 
+# pyre-ignore We don't care about Any here, its not exposed.
+def _match_decorator_unpickler(kwargs: Any) -> "MatchDecoratorMismatch":
+    return MatchDecoratorMismatch(**kwargs)
+
+
 class MatchDecoratorMismatch(Exception):
-    # pyre-ignore We don't care about the type of func, just that its callable.
-    def __init__(self, func: Callable[..., Any], message: str) -> None:
-        super().__init__(
-            # pyre-ignore Pyre doesn't believe functions have __qualname__
-            f"Invalid function signature for {func.__qualname__}: {message}"
```

Comment:
The original code gets `__qualname__` in `MatchDecoratorMismatch` and it looks simpler.
Why move `__qualname__` to callsite? Can we just check the existence of `__qualname__` and fall back when it's not available?

Reply:
Because you can't pickle a `Callable`.

Reply:
What's the use case of pickling MatchDecoratorMismatch? Is it for the output of codemod framework process?

Reply:
Yes. Any exception that can be thrown during a transform will need to be pickled and displayed to the user when codemodding.